src/extension/background-script/connectors/alby.ts

```diff
@@ -247,20 +247,21 @@ export default class Alby implements Connector {
         token: this.config.oAuthToken, // initialize with existing token
       });
 
-      if (this.config.oAuthToken) {
-        try {
-          if (authClient.isAccessTokenExpired()) {
-            const token = await authClient.refreshAccessToken();
-            await this._updateOAuthToken(token.token);
-          }
-          return authClient;
-        } catch (error) {
-          // if auth token refresh fails, the refresh token has probably expired or is invalid
-          // the user will be asked to re-login
-          console.error("Failed to request new auth token", error);
-        }
+      if (!authClient.isAccessTokenExpired()) {
+        return authClient;
       }
 
+      authClient.on("tokenRefreshed", async (token: Token) => {
```

Comment:
you need to register these listeners immediately after creating the authClient. Currently they won't be registered every time